privacyidea/api/lib/utils.py

```diff
@@ -238,21 +238,41 @@ def getLowerParams(param):
     return ret
 
 
-def get_all_params(param, body):
+def get_all_params(request):
     """
-    Combine parameters from GET and POST requests
+    Retrieve all parameters from a request, no matter if these are GET or POST requests
+    or parameters are contained as viewargs like the serial in DELETE /token/<serial>
+
+    :param request: The flask request object
     """
+    param = request.values
+    body = request.data
     return_param = {}
-    for key in param.keys():
-        return_param[key] = unquote(param[key])
-
-    # In case of serialized JSON data in the body, add these to the values.
-    try:
-        json_data = json.loads(to_unicode(body))
-        for k, v in json_data.items():
-            return_param[k] = v
-    except Exception as exx:
-        log.debug("Can not get param: {0!s}".format(exx))
+    if param:
+        log.debug(u"Update params in request {0!s} {1!s} with values.".format(request.method,
+                                                                              request.base_url))
+        # Add the unquoted HTML and form parameters
+        return_param = {key: unquote(value) for (key, value) in param.items()}
+
+    if request.json:
+        log.debug(u"Update params in request {0!s} {1!s} with JSON data.".format(request.method,
+                                                                                 request.base_url))
+        # Add the original JSON data
+        return_param.update(request.json)
+    elif body:
+        # In case of serialized JSON data in the body, add these to the values.
+        try:
+            json_data = json.loads(to_unicode(body))
+            for k, v in json_data.items():
+                return_param[k] = v
+        except Exception as exx:
+            log.debug("Can not get param: {0!s}".format(exx))
```

Comment:
This debug message bugs me for quite some time now, it just doesn't say anything and pops up on every `GET` request.
Maybe we can check if `request.data` exists (i.e. not empty) before trying to parse it.
Or we could parse it only if the `request.method` and/or the mime-type specify that the body should be json-formatted.

Reply:
Will do

Reply:
I checked through all the tests again, you should not see this log message in the future.

The function was heavily redesign. We first check for json.
And since GET-Requets do not contain a body and we check 262 `elif body`.
So actually we really should only hit this debug log, if there is a post request with a badly formatted body.